src/org/omegat/gui/main/ProjectUICommands.java

```diff
@@ -603,49 +606,52 @@ private static void projectOpenImpl(File projectRootFolder) {
                         throw e;
                     }
                 }
-                // team project - non-exist directories could be created from
-                // repo
-                props.autocreateDirectories();
             } else {
-                // not a team project - ask for non-exist directories
+                // not a team project
                 File projectFile = new File(projectRootFolder, OConsts.FILE_PROJECT);
-                while (!props.isProjectValid()) {
-                    // something wrong with the project - display open dialog to
-                    // fix it
-                    ProjectPropertiesDialog prj = new ProjectPropertiesDialog(
-                            Core.getMainWindow().getApplicationFrame(), props, projectFile.getAbsolutePath(),
-                            ProjectPropertiesDialog.Mode.RESOLVE_DIRS);
-                    prj.setVisible(true);
-                    props = prj.getResult();
-                    prj.dispose();
-                    if (props == null) {
-                        // user clicks on 'Cancel'
-                        return;
-                    }
+                if (!props.isProjectValid()) {
+                    do {
+                        // something wrong with the project.
+                        // We display open dialog to fix it.
+                        ProjectPropertiesDialog prj = new ProjectPropertiesDialog(
+                                Core.getMainWindow().getApplicationFrame(), props, projectFile.getAbsolutePath(),
+                                ProjectPropertiesDialog.Mode.RESOLVE_DIRS);
+                        prj.setVisible(true);
+                        props = prj.getResult();
+                        prj.dispose();
+                        if (props == null) {
+                            // user clicks on 'Cancel'
+                            return;
+                        }
+                    } while (!props.isProjectValid());
+                    needToSaveProperties = true;
```

Comment:
We can always set `needToSaveProperties = true` in while loop, because there are only two conditions, one is abort and another is to lead a modification of project properties.